app/javascript/controllers/event_controller.js

```diff
@@ -14,25 +14,8 @@ export default class extends Controller {
   }
 
   correctAggValue() {
-    if (new URLSearchParams(window.location.search).get("agg") == "d" && !this.possibleAggregationValue.includes("d")) {
-      const url = new URL(window.location.href);
-      url.searchParams.set("agg", "w");
-      history.replaceState(null, "", url);
-     } else if(new URLSearchParams(window.location.search).get("agg") == "m" && !this.possibleAggregationValue.includes("m")) {
-      const url = new URL(window.location.href);
-      url.searchParams.set("agg", "w");
-      history.replaceState(null, "", url);
-     } else if(new URLSearchParams(window.location.search).get("agg") == "y" && !this.possibleAggregationValue.includes("m") && !this.possibleAggregationValue.includes("y")) {
-      const url = new URL(window.location.href);
-      url.searchParams.set("agg", "w");
-      history.replaceState(null, "", url);
-     } else if(new URLSearchParams(window.location.search).get("agg") == "y" && !this.possibleAggregationValue.includes("y")) {
-      const url = new URL(window.location.href);
-      url.searchParams.set("agg", "m");
-      history.replaceState(null, "", url);
-     } else {
-       return;
-     }
-    
+    const url = new URL(window.location.href);
+    url.searchParams.set("agg", this.aggregationValue);
+    history.replaceState(null, "", url);
```

Comment:
not sure if "new URLSearchParams(window.location.search)" is needed first or was it only for searching the url to get the agg parameter? seems to be working fine without it

Reply:
I think it's no longer needed. But now, it will add the `agg=` URL param even if there wasn't one before. This is the case if you navigate to a project and don't select anything from the dropdowns (it will just show the default aggregation of day currently). So, it will add the `agg=` value in this case to the URL. Which doesn't really matter, but to more consistent we should first check if there's an existing `agg` param in the URL, and only set it in that case.

You can do this with `new URLSearchParams(window.location.search).get("agg")`.